### PR TITLE
Update licensing info

### LINKED ISSUE #238: Licensing
Don't know if this has been answered but I'm confused by the licensing on this project that's available [here](https://github.com/shadow-maint/shadow/blob/master/COPYING) it says:

> This license has been obsoleted by the change to the BSD-style copyright.

Does this mean this is licensed under BSD and if so what variant of BSD is it ? I don't want to presume. If this is the wrong place can you redirect me to the correct place to ask this question.

### PR DESCRIPTION
**Pull request by hallyn**
Closes #238

Signed-off-by: Serge Hallyn <serge@hallyn.com>

### CODE REVIEW

**Comment by jfhaugh** [comment]
Commenting to get notified on changes.

Lemme know when you're done and I'll review and approve so there's traceability that I approved nuking the Artistic License. If anyone needs a signed piece of paper, I can do that as well.

File: lib/commonio.c
Comment (andrewshadura) [inline lib/commonio.c:5]: How about removing spaces from the year ranges?

Comment (andrewshadura) [reply lib/commonio.c:5]: Hmm, from the rest of the files this seems like a conscious style decision, not an oversight.

**Review by andrewshadura — APPROVED** [review]

File: lib/commonio.c
Comment (hallyn) [reply lib/commonio.c:5]: Yes, I'm maintaining the spacing from the original copyright statements.  Not sure how that fits with spdx specs, but it makes it easier to read IMO.

**Comment by hallyn** [comment]
> Commenting to get notified on changes.
> 
> Lemme know when you're done and I'll review and approve so there's traceability that I approved nuking the Artistic License. If anyone needs a signed piece of paper, I can do that as well.

I'll wait on that approval and this merge before doing a next release.  That way hopefully the release can be used as straightforwardly as possible in the new debian merge.

**Comment by rbalint** [comment]
I'm waiting with the Debian upload until this licensing change lands (and ideally gets in a release).

File: COPYING
Comment (rbalint) [inline COPYING:7]: SPDX

File: COPYING
Comment (rbalint) [inline COPYING:1]: I think BSD-3-Clause is not a valid license for the whole project. I think there is none, since `contrib/atudel` is BSD-4-Clause which is not compatible with GPL https://en.wikipedia.org/wiki/BSD_licenses#4-clause_license_(original_%22BSD_License%22), yet `src/vipw.c` is licensed under GPL.
I suggest dropping `contrib/atudel` or splitting it out to a separate project.

If `vipw.c` stays in the source under GPL, then the the license of the project should be GPL (with BSD parts).

**Review by rbalint — CHANGES_REQUESTED** [review]

File: COPYING
Comment (andrewshadura) [reply COPYING:1]: I don't think so since those are all separate programs, and this situation is far from being unique; I've seen quite a few projects with non-GPL parts, e.g. an application and a client library to talk to it.

Comment (rbalint) [reply COPYING:1]: @andrewshadura Do you have some links? I guess you question the GPL as the whole project's license and not the problem with `atudel`.

Comment (andrewshadura) [reply COPYING:1]: Most definitely GPL is not the project’s license, but just the license of one independent program.

Comment (rbalint) [reply COPYING:1]: @andrewshadura I agree that the project is belived to be BSD licensed, but looking at the current state the release tarballs can't be distributed under BSD. If we remove `atudel` the resulting tarball can be distributed under GPL, but not under BSD (3 clause), because it contains a GPL licensed source file. If we cut `vipw.c`, the remaining tarball can be distributed under BSD (3 clause).
From Debian's perspective I'm OK with keeping `vipw.c` in the tarball, but I have to exclude `atudel`, for sure, to keep `shadow`'s source in `main`.
It could be claimed, that `atudel` and `vipw.c` are only bundled, but this needs an explanation in the COPYING file, IMO.

Comment (andrewshadura) [reply COPYING:1]: @rbalint, you’re misinterpreting it. `atudel` has no relationship whatsoever to `vipw` except being shipped in the same tarball. It is not a derivative work of `vipw`, it’s not based on it, linked with it or anything really except being bundled together.

You cannot claim that the license of the whole package is GPL, because it’s BSD-3-Clause for most files, GPL for `vipw` and BSD-4-Clause for `atudel`. I don’t think a concept of a "license for the whole package" makes sense at all in this case.

File: COPYING
Comment (andrewshadura) [reply COPYING:7]:
```suggestion
2. carry an SPDX header declaring what license applies.
```

Comment (hallyn) [reply COPYING]: Oops.  Thanks.

File: COPYING
Comment (hallyn) [reply COPYING:1]: Originally I'd planned to remove the copyright statement from the files which fall under the 3 clause BSD license.  The COPYING file then just listed the default.  It explicitly says that files may list alternative licenses, which vipw does.

Perhaps I should add a paragraph to say that all new files are to be BSD-3.  But as @andrewshadura says, vipw and atudel are the exceptions.  It may make a difference from a debian packaging perspective, but for the git repo all I'm doing is documenting the existing licenses.  We will not relicense other files just because vipw is under src/.  In particular, section 2 of the GPL says:

```
In addition, mere aggregation of another work not based on the Program with the Program (or with a work based on the Program) on a volume of a storage or distribution medium does not bring the other work under the scope of this License. 
```

The git repo is (in my estimation) a storage or distribution medium.  No other programs in shadow are based on vipw.

**Pull request merged**
